tools/sql/handler.go

```diff
@@ -177,31 +150,57 @@ func doDropDatabase(p ConnectParams, name string) {
 	conn.Close()
 }
 
-func parseConnectParams(cli *cli.Context) (*ConnectParams, error) {
-	params := new(ConnectParams)
-	params.Host = cli.GlobalString(schema.CLIOptEndpoint)
-	params.Port = cli.GlobalInt(schema.CLIOptPort)
-	params.User = cli.GlobalString(schema.CLIOptUser)
-	params.Password = cli.GlobalString(schema.CLIOptPassword)
-	params.Database = cli.GlobalString(schema.CLIOptDatabase)
-	params.DriverName = cli.GlobalString(schema.CLIOptDriverName)
+func parseConnectConfig(cli *cli.Context) (*config.SQL, error) {
+	cfg := new(config.SQL)
+
+	host := cli.GlobalString(schema.CLIOptEndpoint)
+	port := cli.GlobalInt(schema.CLIOptPort)
+	cfg.ConnectAddr = fmt.Sprintf("%s:%v", host, port)
+
+	cfg.User = cli.GlobalString(schema.CLIOptUser)
+	cfg.Password = cli.GlobalString(schema.CLIOptPassword)
+	cfg.DatabaseName = cli.GlobalString(schema.CLIOptDatabase)
+	cfg.DriverName = cli.GlobalString(schema.CLIOptDriverName)
 	isDryRun := cli.Bool(schema.CLIOptDryrun)
-	if err := ValidateConnectParams(params, isDryRun); err != nil {
+
+	if cli.GlobalBool(schema.CLIFlagEnableTLS) {
+		cfg.TLS = &auth.TLS{
+			Enabled:                true,
+			CertFile:               cli.GlobalString(schema.CLIFlagTLSCertFile),
+			KeyFile:                cli.GlobalString(schema.CLIFlagTLSKeyFile),
+			CaFile:                 cli.GlobalString(schema.CLIFlagTLSCaFile),
+			EnableHostVerification: cli.GlobalBool(schema.CLIFlagTLSEnableHostVerification),
+		}
+	}
+
+	if cfg.ConnectAttributes == nil {
+		cfg.ConnectAttributes = map[string]string{}
+	}
+
+	if cli.GlobalString(schema.CLIFlagConnectAttributesTxIsolation) != "" {
+		cfg.ConnectAttributes["tx_isolation"] = cli.GlobalString(schema.CLIFlagConnectAttributesTxIsolation)
+	}
+
+	if err := ValidateConnectConfig(cfg, isDryRun); err != nil {
 		return nil, err
 	}
-	return params, nil
+	return cfg, nil
 }
 
-// ValidateConnectParams validates params
-func ValidateConnectParams(params *ConnectParams, isDryRun bool) error {
-	if len(params.Host) == 0 {
+// ValidateConnectConfig validates params
+func ValidateConnectConfig(cfg *config.SQL, isDryRun bool) error {
```

Comment:
can you also add validation to make sure when TLS is enabled, all required params are also set ?